app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/screen/TeleconsultRecordScreen.kt

```diff
@@ -0,0 +1,146 @@
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.jakewharton.rxbinding3.view.clicks
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.ofType
+import kotlinx.android.synthetic.main.screen_teleconsult_record.view.*
+import org.simple.clinic.R
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.di.injector
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.teleconsultlog.success.TeleConsultSuccessScreenKey
+import org.simple.clinic.teleconsultlog.teleconsultrecord.Answer
+import org.simple.clinic.teleconsultlog.teleconsultrecord.TeleconsultationType
+import org.simple.clinic.teleconsultlog.teleconsultrecord.TeleconsultationType.Audio
+import org.simple.clinic.teleconsultlog.teleconsultrecord.TeleconsultationType.Message
+import org.simple.clinic.teleconsultlog.teleconsultrecord.TeleconsultationType.Video
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.widgets.UiEvent
+import javax.inject.Inject
+
+class TeleconsultRecordScreen(
+    context: Context,
+    attrs: AttributeSet
+) : ConstraintLayout(context, attrs), TeleconsultRecordUi, UiActions {
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  @Inject
+  lateinit var effectHandlerFactory: TeleconsultRecordEffectHandler.Factory
+
+  private val radioIdToTeleconsultationType = mapOf(
+      R.id.teleconsultTypeAudioRadioButton to Audio,
+      R.id.teleconsultTypeVideoRadioButton to Video,
+      R.id.teleconsultTypeMessageRadioButton to Message
+  )
+
+  private val events by unsafeLazy {
+    doneClicks()
+        .compose(ReportAnalyticsEvents())
+  }
+
+  private val screenKey by unsafeLazy {
+    screenRouter.key<TeleconsultRecordScreenKey>(this)
+  }
+
+  private val delegate by unsafeLazy {
+    val uiRenderer = TeleconsultRecordUiRenderer(this)
+
+    MobiusDelegate.forView(
+        events = events.ofType(),
+        defaultModel = TeleconsultRecordModel.create(
+            patientUuid = screenKey.patientUuid,
+            teleconsultRecordId = screenKey.teleconsultRecordId
+        ),
+        init = TeleconsultRecordInit(),
+        update = TeleconsultRecordUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build(),
+        modelUpdateListener = uiRenderer::render
+    )
+  }
+
+  override fun onAttachedToWindow() {
+    super.onAttachedToWindow()
+    delegate.start()
+  }
+
+  override fun onDetachedFromWindow() {
+    delegate.stop()
+    super.onDetachedFromWindow()
+  }
+
+  override fun onSaveInstanceState(): Parcelable? {
+    return delegate.onSaveInstanceState(super.onSaveInstanceState())
+  }
+
+  override fun onRestoreInstanceState(state: Parcelable?) {
+    super.onRestoreInstanceState(delegate.onRestoreInstanceState(state))
+  }
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) return
+
+    context.injector<Injector>().inject(this)
+    backClicks()
```

Comment:
Can we rename this to something like `setupBackClickListener()`?

Reply:
This is a temporary method, it is replaced in the final PR with a Observable stream so that we can pass it to Mobius loop.